server/src/test/server.test.ts

```diff
@@ -16,15 +15,127 @@ import {
+import {
+  collectionDiagnostic,
+  BundleValidationQueue,
+  BundleValidationRequest,
+  DiagnosticCode,
+  validateCollection,
+  validateCollectionModules,
+  validateModule,
+  validateModuleImagePaths,
+  validateModuleLinks
+} from '../bundle-validation'
 import { DOMParser } from 'xmldom'
+import {
+  bundleTreesHandler
+} from '../server-handler'
 
 const DIAGNOSTIC_SOURCE = 'cnxml'
 
 function expect<T>(value: T | null | undefined): T {
   return expectOrig(value, 'test_assertion')
 }
 
+describe('bundleTrees server request', function () {
+  before(function () {
+    mockfs({
+      '/bundle/media/test.jpg': '',
+      '/bundle/collections/invalidslug.xml': `
+        <col:collection xmlns:col="http://cnx.rice.edu/collxml" xmlns:md="http://cnx.rice.edu/mdml">
+          <col:metadata>
+            <md:title>valid</md:title>
+          </col:metadata>
+          <col:content />
+        </col:collection>
+      `,
+      '/bundle/collections/valid.xml': `
+        <col:collection xmlns:col="http://cnx.rice.edu/collxml" xmlns:md="http://cnx.rice.edu/mdml">
+          <col:metadata>
+            <md:slug>valid xml slug</md:slug>
+            <md:title>valid xml title</md:title>
+          </col:metadata>
+          <col:content />
+        </col:collection>
+      `,
+      '/bundle/collections/invalidtitle.xml': `
+      <col:collection xmlns:col="http://cnx.rice.edu/collxml" xmlns:md="http://cnx.rice.edu/mdml">
+        <col:metadata>
+          <md:slug>valid</md:slug>
+        </col:metadata>
+        <col:content />
+      </col:collection>
+    `,
+      '/bundle/modules/valid/index.cnxml': `
+        <document xmlns="http://cnx.rice.edu/cnxml">
+          <metadata xmlns:md="http://cnx.rice.edu/mdml">
+            <md:title>Module</md:title>
+          </metadata>
+          <content />
+        </document>
+      `
+    })
+  })
+  after(function () {
+    mockfs.restore()
+  })
+  it('should handle collection parsing error for xml title', async () => {
+    const connection = {
+      sendDiagnostics: sinon.stub()
+    }
+    const bundle = await BookBundle.from('/bundle')
+    const validationQueue = new BundleValidationQueue(bundle, connection as any)
+    const invalidTitleUri = 'file:///bundle/collections/invalidtitle.xml'
+    const workspaceBookBundles: Map<string, [BookBundle, BundleValidationQueue]> = new Map()
+    workspaceBookBundles.set('file:///bundle', [bundle, validationQueue])
+    const handler = bundleTreesHandler(workspaceBookBundles, connection as any)
+    const request = { workspaceUri: 'file:///bundle' }
+    const bundleTreesResponse = await handler(request)
+    const diagnostic = await collectionDiagnostic()
+    assert(bundleTreesResponse)
+    assert(connection.sendDiagnostics.calledTwice)
+    assert(connection.sendDiagnostics.withArgs({
```

Comment:
I don't think this assertion is doing quite what you want (e.g. if you make `onFirstCall()` `onSecondCall()` here, the assertion will still pass). Perhaps these match your intent in these two test cases?:

```javascript
    assert(connection.sendDiagnostics.getCall(0).calledWithExactly({
      diagnostics: diagnostic,
      uri: invalidSlugUri
    }))
    assert(connection.sendDiagnostics.getCall(1).calledWithExactly({
      diagnostics: diagnostic,
      uri: invalidTitleUri
    }))
```